### DISCUSSION TITLE
Structure for categories representing versions of the software

### DISCUSSION THREAD

**Discussion by joubu** [WeblateOrg/weblate#10112] [General] [answered]
This is a follow-up for  #9556
First, thanks a lot for the "categories" structure. It seems to answer our needs!

However I found several inconsistencies (at least it's how I see things). I am opening this discussion to get feedback on the different points below, I can report them separately with issues if they are confirmed and it's not me misunderstanding how it is supposed to work.

I have the following structure:
project: "koha" > category: "master" > component: "installer"
→ It represents the po file of our installer module for the master branch.
The plan is to have one category per version (23.11, 23.05, etc.) and all our po files under each of them.

1. I created the first component for master, all went well. Now I would like to create another one ("messages") and reuse the same repo (it's not conceivable to have one clone per component). The creation form is not allowing `weblate://koha/master%252Finstaller` how I would have expected, but only `weblate://koha/installer`. Which does not make sense for us, as the "installer" components will have to track different branches.
2. `GET /api/components/koha/master%252Finstaller/'` returns a link to the category `"category": "http://127.0.0.1:8080/api/categories/1/"`. But when I try to reuse it to create a new component using the REST API I get `{"category":["Invalid hyperlink - Object does not exist."]}`
What am I supposed to use?
3. Just a note here, that might be related: `GET /api/components/koha/master%252Finstaller/repository/` returns `"url": "http://127.0.0.1:8080/api/components/koha/installer/repository/",`, but this url returns a 404.

**Comment by joubu** [comment]
4. I have used `wlc` to download translations.
You can download the translations for all the languages: `wlc download koha/master%252Finstaller --output o`
or download a specific file for a given language:`wlc download koha/master%252Finstaller/es --output o`.

But `wlc download koha/master/es --output o` is not working. Should I report it there (wlc), here, or is it a known limitation?

**Comment by nijel** [comment, marked as answer]
1. Repo link were ommitted in the 5.0 release, it was already fixed in d9de3163845769318ea0ec502546c9edc1299939 for 5.1.
2. Bug, fixed in https://github.com/WeblateOrg/weblate/pull/10115
3. Bug, fixed in https://github.com/WeblateOrg/weblate/pull/10114
4. There currently is no such API endpoint.

**Reply by joubu** [reply]
> 1. Repo link were ommitted in the 5.0 release, it was already fixed in [d9de316](https://github.com/WeblateOrg/weblate/commit/d9de3163845769318ea0ec502546c9edc1299939) for 5.1.

Excellent! It's now working with weblate://koha/master/installer indeed.

>     2. Bug, fixed in [api: Fixed creating a categorized component #10115](https://github.com/WeblateOrg/weblate/pull/10115)

Confirmed!

>     3. Bug, fixed in [api: Fixed some API URLs for categorized components #10114](https://github.com/WeblateOrg/weblate/pull/10114)

Tested nijel/api-repo and it's working as expected now.

>     4. There currently is no such API endpoint.

I will fetch them one by one then.

Thanks a lot of your time and your quick fixes, I really appreciate it!
